src/server/scripts/Northrend/Naxxramas/boss_four_horsemen.cpp

```diff
@@ -690,70 +690,77 @@ class boss_four_horsemen_sir : public CreatureScript
+                 case 2:
+                     damage = 500;
+                     break;
+                 case 3:
+                     damage = 1000;
+                     break;
+                 case 4:
+                     damage = 1500;
+                     break;
+                 case 5:
+                     damage = 4000;
+                     break;
+                 case 6:
+                     damage = 12000;
+                     break;
+                 default:
+                     damage = 20000 + 1000 * (GetStackAmount() - 7);
+                     break;
+             }
+             if (damage)
+             {
+                 CastSpellExtraArgs args(TRIGGERED_FULL_MASK);
+                 args.AddSpellBP0(damage);
+                 caster->CastSpell(GetTarget(), SPELL_MARK_DAMAGE, args);
+             }
+         }
+     }
+
+     void Register() override
+     {
+         AfterEffectApply += AuraEffectApplyFn(spell_four_horsemen_mark::OnApply, EFFECT_0, SPELL_AURA_DUMMY, AURA_EFFECT_HANDLE_REAL_OR_REAPPLY_MASK);
+     }
+};
 
-            void OnApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
-            {
-                if (Unit* caster = GetCaster())
-                {
-                    int32 damage;
-                    switch (GetStackAmount())
-                    {
-                        case 1:
-                            damage = 0;
-                            break;
-                        case 2:
-                            damage = 500;
-                            break;
-                        case 3:
-                            damage = 1000;
-                            break;
-                        case 4:
-                            damage = 1500;
-                            break;
-                        case 5:
-                            damage = 4000;
-                            break;
-                        case 6:
-                            damage = 12000;
-                            break;
-                        default:
-                            damage = 20000 + 1000 * (GetStackAmount() - 7);
-                            break;
-                    }
-                    if (damage)
-                    {
-                        CastSpellExtraArgs args(TRIGGERED_FULL_MASK);
-                        args.AddSpellBP0(damage);
-                        caster->CastSpell(GetTarget(), SPELL_MARK_DAMAGE, args);
-                    }
-                }
-            }
+ class spell_four_horsemen_consumption : public SpellScript
+ {
+     PrepareSpellScript(spell_four_horsemen_consumption);
 
-            void Register() override
-            {
-                AfterEffectApply += AuraEffectApplyFn(spell_four_horsemen_mark_AuraScript::OnApply, EFFECT_0, SPELL_AURA_DUMMY, AURA_EFFECT_HANDLE_REAL_OR_REAPPLY_MASK);
-            }
-        };
+     void HandleDamageCalc(SpellEffIndex /*effIndex*/)
+     {
+         SetHitDamage(2750);
```

Comment:
not best check if is heroic first?
like:

```cpp
if (GetCaster()->GetMap()->IsHeroic())
    SetHitDamage(4250);
else
    SetHitDamage(2750);
```
instead call SetHitDamage twice?

Reply:
Could it be better like this?
```c++
SetHitDamage((GetCaster()->GetMap()->IsHeroic()) ? 4250 : 2750);
```

Reply:
is fine to me too.
But still is better put it in a variable first.
like:

```cpp
uint32 damage = GetCaster()->GetMap()->IsHeroic() ? 4250 : 2750;
SetHitDamage(damage);
```

Reply:
But in this case with uint16 it's enough 😄

Reply:
yeah, but we always use int32/uint32 to represent damages xD

Reply:
Fixed. Thanks!